pkg/provider/azure_loadbalancer.go

```diff
@@ -919,11 +921,14 @@ func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service,
 
 func (az *Cloud) findMatchedPIPByLoadBalancerIP(service *v1.Service, loadBalancerIP, pipResourceGroup string, pips *[]network.PublicIPAddress) (*network.PublicIPAddress, error) {
 	if pips == nil {
+		// this should not happen
+		return nil, fmt.Errorf("findMatchedPIPByLoadBalancerIP: nil pip list passed")
+	} else if *pips == nil {
```

Comment:
Should it check the length? Empty slice does not equal `nil`.
```suggestion
	} else if len(*pips) == 0 {
```

Reply:
If length is 0, that means we already listed pip but there isn't any. We check nil to decide whether to list pip.

Reply:
> We check nil to decide whether to list pip.
should it be `if pip == nil`?

Reply:
nil is a valid slice, and by convention we will use `nil` as an `empty slice`. I suggest we better don't distinguish them.

In this case, the type is `*[]T`, so why not:
- `pip == nil`: to list pip
- `len(*pip) == 0`: already list, but it's empty
- else

Reply:
if so, could you please add some comments to the function in case the caller misuse `nil` and `[]T`?

Reply:
Added comments to make it clear. Thanks